test/suites/helpers.js

```diff
@@ -223,6 +223,26 @@ describe('Helpers', function() {
         [{name: 'John'}, {name: 'Paul'}]
       );
     });
+
+    describe('Issue #472 - tree/cursor.splice does not conform with the specification as of ES6 (ECMAScript 2015)', function () {
+      it('should be possible to splice an array when omitting the nb (deleteCount) argument', function () {
+        const array = [0, 1, 2, 3, 4];
+
+        assert.deepEqual(splice(array, 2), [0, 1]);
+
+        assert.deepEqual(splice(array, -2), [0, 1, 2]);
+
+        assert.deepEqual(splice(array, 2, null), [0, 1, 2, 3, 4]);
+      });
+
+      it('should delete no values when supplying an argument which is not parseable as number', function () {
```

Comment:
This depends entirely on you, or rather how the splice description is phrased in the readme. Currently it states that splice behaves as per specification. The current implementation of just taking the input and deleting nothing if the deleteCount argument can not get parsed would be the result I'd expect if I read the readme. We could also change the readme to state that unparseable input throws errors.

Normally I'm a friend of the fail-fast principle, according to which we should throw an error here.

However, Baobab is mainly your project, so I leave this decision up to you.